lib/services/database/user_repository_service.dart

```diff
@@ -34,6 +34,15 @@ class UserRepositoryService {
 
     return docSnap.exists;
   }
+
+  /// This method will check if the unique username [username] is already taken
+  /// by some user
+  Future<bool> isUsernameTaken(String username) async {
+    final query = await _collectionRef
+        .where(UserData.usernameField, isEqualTo: username)
```

Comment:
Small suggestion, I think that we could limit the number of elements queried here.
In theory, we should have at most one but if for some reason we have a bug on the db and that a lot of accounts have the same username, limitting the query to 1 could avoid some reading cost.

```suggestion
        .where(UserData.usernameField, isEqualTo: username)
        .limit(1)
```